Feat/on 77 populate case details

tongzxc opened the pull request.

Commit:
feat: create /v1/beneficiaries/:id/cases and get cases from id

create /v1/beneficiaries/:id/cases
get caseNumbers and return beneficiaryCases obj

**RELATES TO**: ON-77

Commit:
add error handling for getBeneficiaryCases

farablanco commented on src/controllers/routers/beneficiaries.js:
```diff
@@ -126,6 +129,45 @@ const getBeneficiary = async (req, res, next) => {
+ * @param {Request} req
+ * @param {Response} res
+ */
+const getBeneficiaryCases = async (req, res, next) => {
```
since the table we're accessing is cases, move this to the cases controller file.
the route is okay. the route should be in beneficiaries.

rename to getCasesByBeneficiaryId

tongzxc replied:
moved getCasesByBeneficiaryId from controllers > beneficiary to controllers > cases
update routers > beneficiary to access cases.getCasesByBeneficiaryId

farablanco commented on src/controllers/routers/beneficiaries.js:
```diff
@@ -126,6 +129,45 @@ const getBeneficiary = async (req, res, next) => {
+  const { id } = req.params;
+  try {
+    // get list of case numbers from Case table based on beneficiaryId
+    const caseNumber = await Case.query()
```
update the var name to avoid having both caseNumber and caseNumbers.
you could change this to results.

tongzxc replied:
changed var name from caseNumber to results

farablanco commented on src/controllers/routers/beneficiaries.js:
```diff
@@ -126,6 +129,45 @@ const getBeneficiary = async (req, res, next) => {
+      caseNumbers.push(number);
+    }
+
+    const ben = await Beneficiary.query()
```
please change ben to beneficiary

tongzxc replied:
changed ben to beneficiary

farablanco reviewed (changes requested):
please see comments

Commit:
feat: update PR

move getCasesByBeneficiaryId from controllers>beneficiary to controllers>cases
update routers>beneficiary to access cases.getCasesByBeneficiaryId
update var names in getCasesByBeneficiaryId

**RELATES TO**: ON-77

Commit:
Merge branch 'develop' of https://github.com/TechLadies/onesg-app into feat/on-77-populate-case-details

Commit:
removed isLoggedIn

farablanco reviewed (approved):
lgtm
